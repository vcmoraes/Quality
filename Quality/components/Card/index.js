import React from 'react'
import styled from 'styled-components'
import { Card } from 'native-base'

const CardView = styled(Card)`
  width: ${props => props.width};
  border-radius: ${props => props.borderRadius};
  padding-top: ${props => props.paddingTop};
  padding-left: ${props => props.paddingLeft};
  padding-right: ${props => props.paddingRight};
  padding-bottom: ${props => props.paddingBottom};
  margin-top: ${props => props.marginTop};
  margin-left: ${props => props.marginLeft};
  margin-right: ${props => props.marginRight};
  margin-bottom: ${props => props.marginBottom};
  justify-content: ${props => props.justifyContent};
  align-items: ${props => props.alignItems};
  elevation: ${props => props.elevation};
  background-color: ${props => props.backgroundColor};
`
CardView.defaultProps = {
  justifyContent: 'flex-start',
  alignItems: 'stretch',
  borderRadius: 10,
  paddingTop: 0,
  paddingLeft: 0,
  paddingRight: 0,
  paddingBottom: 0,
  marginTop: 0,
  marginLeft: 0,
  marginRight: 0,
  marginBottom: 0,
  backgroundColor: 'white',
  elevation: 5,
  width: '100%'
}

export default props => <CardView {...props} />
