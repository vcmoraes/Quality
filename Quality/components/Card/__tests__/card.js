import React from 'react'
import Card from '../index'
import renderer from 'react-test-renderer'

describe('Components - Card', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
