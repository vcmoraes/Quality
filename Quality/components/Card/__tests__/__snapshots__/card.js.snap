// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Card renders correctly 1`] = `
<View
  alignItems="stretch"
  backgroundColor="white"
  borderRadius={10}
  elevation={5}
  justifyContent="flex-start"
  marginBottom={0}
  marginLeft={0}
  marginRight={0}
  marginTop={0}
  paddingBottom={0}
  paddingLeft={0}
  paddingRight={0}
  paddingTop={0}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "alignItems": "stretch",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 5,
        "justifyContent": "flex-start",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
  width="100%"
/>
`;
